ospd/ospd.py

```diff
@@ -1045,18 +1047,25 @@ def handle_get_vts_command(self, vt_et):
             return simple_response_str('get_vts', 404, text)
 
         filtered_vts = None
-        if vt_filter:
+        if not vt_id and vt_filter:
             filtered_vts = self.vts_filter.get_filtered_vts_list(
                 self.vts, vt_filter
             )
+        elif vt_id:
+            filtered_vts = vt_id
+        else:
+            filtered_vts = self.vts.keys()
 
-        responses = []
-
-        vts_xml = self.get_vts_xml(vt_id, filtered_vts)
+        yield xml_helper.create_response('get_vts')
+        yield xml_helper.create_element('vts')
 
-        responses.append(vts_xml)
+        for vt in self.get_vt_iterator():
+            if list(vt.keys())[0] not in filtered_vts:
```

Comment:
This doesn't look got to me. I am really nor sure what this is supposed to do.

If I understand the code correctly it returns the first key of a dict as the vt_id which wont work for Python 3.5 if the dict contains several keys.

It would be better to expect a specific shape of a dict always having a vt_id key like `{'vt_id': ..., }`